lib/db/schema/patch-067-068.sql

```diff
@@ -0,0 +1,155 @@
+ALTER TABLE `unverifiedTokens`
+ADD COLUMN `shortCodeHash` BINARY(32) DEFAULT NULL,
+ALGORITHM = INPLACE, LOCK = NONE;
+
+CREATE INDEX `unverifiedTokens_shortCodeHash`
+ON `unverifiedTokens` (`shortCodeHash`)
+ALGORITHM = INPLACE LOCK=NONE;
+
+ALTER TABLE `unverifiedTokens`
+ADD COLUMN `shortCodeExpiresAt` BIGINT UNSIGNED DEFAULT NULL,
+ALGORITHM = INPLACE, LOCK = NONE;
+
+CREATE PROCEDURE `verifyTokenShortCode_1` (
+  IN `shortCodeHashArg` BINARY(32),
+  IN `uidArg` BINARY(16)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  SET @tokenVerificationId = NULL;
+  SELECT tokenVerificationId INTO @tokenVerificationId FROM unverifiedTokens WHERE uid = uidArg AND shortCodeHash = shortCodeHashArg;
+
+  IF @tokenVerificationId IS NOT NULL THEN
+    SET @updateCount = 0;
+    SELECT COUNT(*) INTO @updateCount FROM unverifiedTokens WHERE tokenVerificationId = @tokenVerificationId  AND shortCodeExpiresAt < (UNIX_TIMESTAMP(NOW(3)) * 1000);
+    IF @updateCount > 0 THEN
+      SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 2101, MESSAGE_TEXT = 'Expired shortcode.';
+    END IF;
+  END IF;
```

Comment:
If we expect the common case to be using a valid, unexpired shortCode, then we should optimize for doing a single read in that case.  I think we can achieve that by having the first query include the expiry check, like:

```
SELECT tokenVerificationId INTO @tokenVerificationId FROM unverifiedTokens WHERE uid = uidArg AND shortCodeHash = shortCodeHashArg and shortCodeExpiresAt >= (UNIX_TIMESTAMP(NOW(3)) * 1000)
```

And then, if we didn't find one, we can whether it exists but is expired with a separate query like above.